### DISCUSSION TITLE
Not able to use custom board for Twister

### DISCUSSION THREAD

**Discussion by thomasBisseling** [zephyrproject-rtos/zephyr#67674] [Q&A]
When testing with twister is get the error `platform_filter - unrecognized platform - <board_name>`

The folder structure is like this:
```tree
├─ project_name
└── boards/arm/<board_name>
    └── <board_name>_defconfig
    └── <board_name>.dts
    └── <board_name>.yaml
```

In YAML is have the `identifier` exact the same as <board_name>.

I run command: `west twister -n -p <board_name> -T <path_to_tests> -A <absolute_path>/<project_name>/boards/ -v`.

In `build.log` is a line with `No board named '<board_name>' found.`

Please help, how do i fix this?

**Comment by nordicjm** [comment]
Did you add `board_root` to your `zephyr/module.yml` file?

**Reply by thomasBisseling** [reply]
No i didn't. But i found out when i copy the folder of my board to  `~/zephyrproject/zephyr/boards/arm`. The test will run. Only this is not the way to do it

**Reply by nordicjm** [reply]
https://docs.zephyrproject.org/latest/develop/application/index.html#boards
https://docs.zephyrproject.org/latest/develop/modules.html#build-settings

**Reply by lexologe** [reply]
Created a zephyr/module.yml with the following content:
```
build:
  cmake: .
tests:
  - tests
boards:
  - boards
```
But it does not work. It creates the `./twister-out/<my-board-name>` directory and in the build.log within that directory it shows:
No board named '`<my-board-name>`' found.

**Reply by lexologe** [reply]
Got over it by removing `[..]/boards/` at the end of the `-A` parameter. Now twister complains about SchemaErrors in custom DTS files. Will create a issue for that.

**Reply by nordicjm** [reply]
> Created a zephyr/module.yml with the following content:
> 
> ```
> build:
>   cmake: .
> tests:
>   - tests
> boards:
>   - boards
> ```
> 
> But it does not work. It creates the `./twister-out/<my-board-name>` directory and in the build.log within that directory it shows: No board named '`<my-board-name>`' found.

This isn't even correct, look at https://github.com/zephyrproject-rtos/example-application/blob/main/zephyr/module.yml
